utils/navigate.py

```diff
@@ -0,0 +1,54 @@
+from utils.log import logger
+
+
+class Navigate(object):
+    def __init__(self):
+        self.cache = {}
+
+    def register(self, cls):
+        def f(obj):
+            if cls not in self.cache:
+                self.cache[cls] = {}
+            self.cache[cls][obj.__name__] = obj
+        return f
+
+    def navigate(self, cls, name):
```

Comment:
this should take cls_or_obj and always do `cls = type(cls_or_obj) if not isinstance(cls_or_obj, type) else cls_or_obj`